### DISCUSSION TITLE
How to use without Docker

### DISCUSSION THREAD

**Discussion by rabol** [NiclasvanEyk/initializer-for-laravel#238] [Q&A]
Hi

Is Docker a hard requirement ?

**Comment by NiclasvanEyk** [comment]
Yes, kind of.

The application contained in the zip file is currently preconfigured for the usage with Docker. This guarantees, that the `initialize` installer script will succeed regardless of your local setup, as long as Docker is installed. 

However, since Laravel is configured using environment variables, you are free to simply delete the `initialize` script and setup and configure the application by yourself. This definitely requires more knowledge and you won't get as much value out of Initializer for Laravel. You can use the steps in the `initialize` script as a guideline.

Now that I think about it, I could  make the usage without Docker more convenient. Right now the `initialize` script starts the Docker containers and then tells them to run commands (via `sail artisan myCommand` or `sail npm install`). 

Instead, I could divide the setup into two scripts, one that boots the containers and one that runs the necessary installation steps. The latter would assume, that the application is configured correctly, and that `php`, `npm` and your database are available. If you don't want to use Docker, you could just run the second one, after adjusting your local `.env` file.

Is this something you would be interested in? Also what are your reasons for not using Docker?

**Reply by rabol** [reply]
I use Laravel Valet
